cross-project-tests/debuginfo-tests/dexter/dex/debugger/DAP.py

```diff
@@ -635,21 +638,32 @@ def launch(self, cmdline):
     def _post_step_hook(self):
         """Hook to be executed after completing a step request."""
 
-    def step_in(self):
+    def _step(self, step_request_string):
         self._flush_breakpoints()
         step_req_id = self.send_message(
-            self.make_request("stepIn", {"threadId": self._debugger_state.thread})
+            self.make_request(
+                step_request_string, {"threadId": self._debugger_state.thread}
+            )
         )
         response = self._await_response(step_req_id)
         if not response["success"]:
-            raise DebuggerException("failed to step")
+            raise DebuggerException(f"failed to {step_request_string}")
```

Comment:
Slight nit, this message is good if we have stepIn or stepOut, but it may be slightly confusing if a user just gets the exception "failed to next" - maybe make it a little more explicit, like "failed to perform debugger action: '{step_request_string}'"?

Reply:
Agreed, done